File: src/main/java/eu/codetopic/utils/ui/activity/modular/SimpleActivityCallBackModule.java
/*
 * Copyright 2017 Jiří Kuchyňka (Anty)
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package eu.codetopic.utils.ui.activity.modular;

import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.Toolbar;
import android.view.KeyEvent;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

public class SimpleActivityCallBackModule extends ActivityCallBackModule {

    private static final String LOG_TAG = "SimpleActivityCallBackModule";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
    }

    @Override
    protected void onNewIntent(Intent intent) {
    }

    @Override
    protected void onSetContentView(@LayoutRes int layoutResID, SetContentViewCallBack callBack) {
        callBack.pass();
    }

    @Override
    protected void onSetContentView(View view, SetContentViewCallBack callBack) {
        callBack.pass();
    }

    @Override
    protected void onSetContentView(View view, ViewGroup.LayoutParams params, SetContentViewCallBack callBack) {
        callBack.pass();
    }

    @Override
    protected Toolbar onSetSupportActionBar(@Nullable Toolbar toolbar) {
        return toolbar;
    }

    @Override
    protected void onPostSetSupportActionBar(@Nullable Toolbar toolbar) {
    }

    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
    }

    @Override
    protected void onStart() {
    }

    @Override
    protected void onResume() {
    }

    @Override
    protected boolean onCreateOptionsMenu(Menu menu) {
        return false;
    }

    @Override
    protected boolean onOptionsItemSelected(MenuItem item) {
        return false;
    }

    @Override
    protected void onTitleChanged(CharSequence title, int color) {
    }

    @Override
    protected void onConfigurationChanged(Configuration newConfig) {
    }

    @Override
    protected void onBackPressed() {
    }

    @Override
    protected boolean onKeyDown(int keyCode, KeyEvent event) {
        return false;
    }

    @Override
    protected void onAttachFragment(Fragment fragment) {
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
    }

    @Override
    protected void onPause() {
    }

    @Override
    protected void onStop() {
    }

    @Override
    protected void onDestroy() {
    }

    @Override
    protected void onRestart() {
    }
}

File: src/main/java/eu/codetopic/utils/ui/activity/modular/SimpleActivityCallBackModule.kt
/*
 * utils
 * Copyright (C)   2017  anty
 *
 * This program is free  software: you can redistribute it and/or modify
 * it under the terms  of the GNU General Public License as published by
 * the Free Software  Foundation, either version 3 of the License, or
 * (at your option) any  later version.
 *
 * This program is distributed in the hope that it  will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied  warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
 * GNU General Public License for more details.
 *
 * You  should have received a copy of the GNU General Public License
 * along  with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package eu.codetopic.utils.ui.activity.modular

import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.support.annotation.LayoutRes
import android.support.v4.app.Fragment
import android.support.v7.widget.Toolbar
import android.view.KeyEvent
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup

open class SimpleActivityCallBackModule : ActivityCallBackModule() {

    companion object {

        private const val LOG_TAG = "SimpleActivityCallBackModule"
    }

    override fun onCreate(savedInstanceState: Bundle?) {}

    override fun onNewIntent(intent: Intent) {}

    override fun onSetContentView(@LayoutRes layoutResID: Int,
                                  callBack: ActivityCallBackModule.SetContentViewCallBack) {
        callBack.pass()
    }

    override fun onSetContentView(view: View,
                                  callBack: ActivityCallBackModule.SetContentViewCallBack) {
        callBack.pass()
    }

    override fun onSetContentView(view: View, params: ViewGroup.LayoutParams,
                                  callBack: ActivityCallBackModule.SetContentViewCallBack) {
        callBack.pass()
    }

    override fun onSetSupportActionBar(toolbar: Toolbar?): Toolbar? = toolbar

    override fun onPostSetSupportActionBar(toolbar: Toolbar?) {}

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {}

    override fun onStart() {}

    override fun onResume() {}

    override fun onCreateOptionsMenu(menu: Menu): Boolean = false

    override fun onOptionsItemSelected(item: MenuItem): Boolean = false

    override fun onTitleChanged(title: CharSequence, color: Int) {}

    override fun onConfigurationChanged(newConfig: Configuration) {}

    override fun onBackPressed() {}

    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean = false

    override fun onAttachFragment(fragment: Fragment) {}

    override fun onSaveInstanceState(outState: Bundle) {}

    override fun onPause() {}

    override fun onStop() {}

    override fun onDestroy() {}

    override fun onRestart() {}
}
